### ISSUE TITLE
getLocale doesn't detect locale for format: "file" for /index.astro

### ISSUE THREAD

**Issue by johnlife** [jlarmstrongiv/astro-i18n-aut#27]
const locale = getLocale(Astro.url);

incorrectly detects the default locale for index.astro in setup like this:

```
  trailingSlash: "never",
  build: {
    format: "file",
  },
```

since locale urls are 
/fr.html  

the code at https://github.com/jlarmstrongiv/astro-i18n-aut/blob/a68189f7e58e84c4d4605f70a614af2ee0304ee6/src/edge-runtime/getLocale.ts#L46-L48 fails to detect it as `fr` locale

**Comment by johnlife** [comment]
Here you go:
https://github.com/johnlife/naut-reproduction

notice the `yarn build` output:

```
C:\projects\evenito\temp\naut0reproduction>yarn build
yarn run v1.22.19
warning package.json: No license field
$ astro build
11:28:44 AM [content] No content directory found. Skipping type generation.
11:28:44 AM [build] output target: static
11:28:44 AM [build] Collecting build info...
11:28:44 AM [build] Completed in 92ms.
11:28:44 AM [build] Building static entrypoints...
11:28:45 AM [build] Completed in 635ms.

 generating static routes
▶ src/astro_tmp_pages_en/index.astro
/en.html en
  └─ /en.html (+15ms)
▶ src/astro_tmp_pages_es/index.astro
/es.html en
  └─ /es.html (+4ms)
▶ src/astro_tmp_pages_fr/index.astro
/fr.html en
  └─ /fr.html (+3ms)
▶ src/pages/index.astro
/ en

```

Notice that for `/es.html` and for `fr.html` the detected locale was still `en`

**Comment by jlarmstrongiv** [comment]
@johnlife thank you for the reproduction and the extra information!  That helps a lot.  I’ll re-open and get to it as soon as I can (family in town this week)

**Comment by jlarmstrongiv** [comment]
After investigating, I've found that Astro includes the `.html` in the pathname during the static build, which breaks the locale helpers with these particular settings.  I will fix it on Tuesday 😄

**Comment by jlarmstrongiv** [comment]
Fixed in `0.4.24` 🎉 thank you so much for your help with the reproduction

I patched all the helper functions to account for trailing `.html` in the url pathname
